src/main/java/core/mate/academy/service/MachineService.java

```diff
@@ -1,38 +1,13 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Machine;
 import java.util.List;
 
-/**
- * Parameterize this service and add its implementation in a separate class.
- */
-public interface MachineService {
+public interface MachineService<T extends Machine> {
```

Comment:
If we generify our interface like this, it is dependent on the exact implementation (`MachineServiceImpl`), which works only with `Machine` hierarchy. Therefore, this particular interface is not reusable for any other type
```suggestion
public interface MachineService<T> {
```

Reply:
![image](https://user-images.githubusercontent.com/58385506/102114200-9bf1a280-3e42-11eb-8398-6bb9c12b9615.png)